### PR TITLE
[next] Ensure we do not include ending slash in matched path

### PR DESCRIPTION
**Pull request by ijjk**
We don't expect a trailing slash to be included in this header so don't include it when matching. 

x-ref: [slack thread](https://vercel.slack.com/archives/C0676QZBWKS/p1720633898241439)

### CODE REVIEW

**Review by sueplex — DISMISSED** [review]
LGTM

File: packages/next/test/fixtures/00-trailing-slash-middleware-add/middleware.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+      export default function middleware() {
```
Comment (ztanner) [inline packages/next/test/fixtures/00-trailing-slash-middleware-add/middleware.js]: nit: lint

**Review by ztanner — DISMISSED** [review]

File: packages/next/test/fixtures/00-trailing-slash-middleware-add/middleware.js (hunk 1 above)
Comment (ijjk) [reply packages/next/test/fixtures/00-trailing-slash-middleware-add/middleware.js]: oh this file is written dynamically so can be removed actually.

**Review by ztanner — APPROVED** [review]

**Review by trek — APPROVED** [review]

**Comment by EndangeredMassa** [comment]
@ijjk are these tests legit to the code change?

**Comment by ijjk** [comment]
@EndangeredMassa no none of them appear Next.js related or related to this change at all. They were passing before we merged main.

**Pull request merged**